src/visitors/sympy_solver_visitor.cpp

```diff
@@ -424,47 +398,25 @@ void SympySolverVisitor::visit_diff_eq_expression(ast::DiffEqExpression& node) {
     check_expr_statements_in_same_block();
 
     const auto node_as_nmodl = to_nmodl_for_sympy(node);
-    const auto locals = py::dict("equation_string"_a = node_as_nmodl,
-                                 "dt_var"_a = codegen::naming::NTHREAD_DT_VARIABLE,
-                                 "vars"_a = vars,
-                                 "use_pade_approx"_a = use_pade_approx,
-                                 "function_calls"_a = function_calls);
-
+    auto diffeq_solver =
+        nmodl::pybind_wrappers::EmbeddedPythonLoader::get_instance().api()->create_des_executor();
+    diffeq_solver->node_as_nmodl = node_as_nmodl;
+    diffeq_solver->dt_var = codegen::naming::NTHREAD_DT_VARIABLE;
+    diffeq_solver->vars = vars;
+    diffeq_solver->use_pade_approx = use_pade_approx;
+    diffeq_solver->function_calls = function_calls;
+    diffeq_solver->method = solve_method;
+    (*diffeq_solver)();
     if (solve_method == codegen::naming::EULER_METHOD) {
-        logger->debug("SympySolverVisitor :: EULER - solving: {}", node_as_nmodl);
         // replace x' = f(x) differential equation
         // with forwards Euler timestep:
         // x = x + f(x) * dt
```

Comment:
nitpick : as these are now in `diffeq_solver`, they can babe removed?

Reply:
I'm keeping these branches for logging - to not lose this behavior. the python wrappers I want to keep free of other code, it should be really the minimum glue for running the python code. But I agree it's not very satisfactory...